### PR TITLE
PYTHON-2850 [v3.13] Deprecate mapReduce command

### PR DESCRIPTION
**Pull request by juliusgeo**
Deprecate mapReduce helpers

### CODE REVIEW

File: pymongo/collection.py [hunk 1]
```diff
@@ -3091,6 +3093,10 @@ def map_reduce(self, map, reduce, out, full_response=False, session=None, **kwar
            this collection is automatically applied to this operation (if the
            output is not inline) when using MongoDB >= 3.4.
 
+        .. versionchanged:: 3.13
+           Deprecated :meth:`map_reduce`. Support for this function is deprecated in
```
Comment (behackett) [inline pymongo/collection.py:3097]: We should also say that this is removed in PyMongo 4.0:
https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#collection-map-reduce-and-collection-inline-map-reduce-are-removed

You should say here what users should do instead. An abbreviated version of what the PyMongo 4 upgrade guide says.

File: pymongo/collection.py [hunk 2]
```diff
@@ -3153,11 +3163,16 @@ def inline_map_reduce(self, map, reduce, full_response=False, session=None, **kw
 
             >>> db.test.inline_map_reduce(map, reduce, limit=2)
 
+        .. versionchanged:: 3.13
+           Deprecated :meth:`inline_map_reduce`. Support for this function is deprecated in
```
Comment (behackett) [inline pymongo/collection.py:3167]: Same comment here.

**Review by behackett — CHANGES_REQUESTED** [review]

**Comment by juliusgeo** [comment]
Managed to accidentally merge with master rather than the v3.13 branch so there are a tonnn of messed up changes. I think I'm just gonna create this from scratch in a new PR given how small it is.

**Review by blink1073 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**